### PR TITLE
Drop `git ls-files` in gemspec

### PR DESCRIPTION
**Pull request by utkarsh2102**
Hi,

Thanks for working on this! :heart: 
However, while maintaining this in Debian, we found that this library relies on `git` to list the files which could be done via pure Ruby alternative -- which is what this PR does.

As an addition, this PR makes sure that this gem only ships the required files to the end-users and not other things which are not needed by them! :rocket: 

Also, added `rubocop-packaging` as a `development_dependency` which will ensure the best practices.
Here's what it shows:

```
Inspecting 172 files
...C..........C.........C...........
....................................
....................................
....................................
............................

Offenses:

aruba.gemspec:48:16: C: Packaging/GemspecGit: Avoid using git to produce lists of files.
Downstreams often need to build your package in an environment that does not have git (on purpose).
Use some pure Ruby alternative, like Dir or Dir.glob.
  spec.files = `git ls-files -z`.split("\x0").reject do |f|
               ^^^^^^^^^^^^^^^^^
fixtures/cli-app/cli-app.gemspec:18:24: C: Packaging/GemspecGit: Avoid using git to produce lists of files.
Downstreams often need to build your package in an environment that does not have git (on purpose).
Use some pure Ruby alternative, like Dir or Dir.glob.
  spec.files         = `git ls-files -z`.split("\x0")
                       ^^^^^^^^^^^^^^^^^
fixtures/empty-app/cli-app.gemspec:18:24: C: Packaging/GemspecGit: Avoid using git to produce lists of files.
Downstreams often need to build your package in an environment that does not have git (on purpose).
Use some pure Ruby alternative, like Dir or Dir.glob.
  spec.files         = `git ls-files -z`.split("\x0")
                       ^^^^^^^^^^^^^^^^^

172 files inspected, 3 offenses detected
```

And this PR fixes the same, which helps us in shipping this in Debian! :tada: 
Hope this would make sense and you'll be open to this change :100: 

Signed-off-by: Utkarsh Gupta <<utkarsh@debian.org>>

---

CC: @luke-hill 
Closes: #720

### CODE REVIEW

File: aruba.gemspec [hunk 1]
```diff
@@ -45,11 +46,9 @@ Gem::Specification.new do |spec|
   spec.rubygems_version = '>= 1.6.1'
   spec.required_ruby_version = '>= 2.4'
 
-  spec.files = `git ls-files -z`.split("\x0").reject do |f|
-    f.match(%r{^(test|spec|features)/})
-  end
+  spec.files = Dir['{docs,fixtures,lib}/**/*', 'CHANGELOG.md', 'LICENSE', 'README.md']
```
Comment (mvz) [inline aruba.gemspec:49]: Please add a `.exclude(*File.read('.gitignore').split)` like you did in for cucumber-rails and gir_fii.

Comment (utkarsh2102) [reply aruba.gemspec:49]: Aw, crap. Right!

Comment (utkarsh2102) [reply aruba.gemspec]: Should've done earlier but done now!
Thanks for the review \o/

File: aruba.gemspec [hunk 2]
```diff
@@ -46,7 +46,8 @@ Gem::Specification.new do |spec|
   spec.rubygems_version = '>= 1.6.1'
   spec.required_ruby_version = '>= 2.4'
 
-  spec.files = Dir['{docs,fixtures,lib}/**/*', 'CHANGELOG.md', 'LICENSE', 'README.md']
+  spec.files = Rake::FileList['{docs,fixtures,lib}/**/*', 'CHANGELOG.md', \
```
Comment (olleolleolle) [inline aruba.gemspec:49]: This PR now does not have the explicit `require 'rake/file_list'` - is that significant?

Comment (utkarsh2102) [reply aruba.gemspec:50]: Eeks. Thanks, @olleolleolle. Fixed! :heart:

**Pull request merged**